var/spack/repos/builtin/packages/py-pygobject/package.py

```diff
@@ -25,30 +25,48 @@
 from spack import *
 
 
-class PyPygobject(AutotoolsPackage):
+class PyPygobject(PythonPackage):
     """bindings for the GLib, and GObject,
        to be used in Python."""
 
     homepage = "https://pypi.python.org/pypi/pygobject"
 
     url      = "http://ftp.gnome.org/pub/GNOME/sources/pygobject/2.28/pygobject-2.28.6.tar.bz2"
 
+    version('3.28.3', 'c70b2d8c492853fca5d7fd556ce39d36',
+            url='https://files.pythonhosted.org/packages/e0/e8/1e4f21800015a9ca153969e85fc29f7962f8f82fc5dbc1ecbdeb9dc54c75/PyGObject-3.28.3.tar.gz')
     version('2.28.6', 'a43d783228dd32899e6908352b8308f3')
     version('2.28.3', 'aa64900b274c4661a5c32e52922977f9')
 
     extends('python')
 
-    depends_on('pkgconfig', type=('build'))
+    depends_on('py-setuptools', type=('build'))
     depends_on("libffi")
     depends_on('glib')
-    depends_on('py-py2cairo', type=('build', 'run'))
+    depends_on('python@2:2.99', when='@2:2.99')
+    depends_on('py-pycairo', type=('build', 'run'), when='@3:')
+    depends_on('py-py2cairo', type=('build', 'run'), when='@2:2.99')
     depends_on('gobject-introspection')
+    depends_on('gtkplus', when='@3:')
 
     patch('pygobject-2.28.6-introspection-1.patch', when='@2.28.3:2.28.6')
 
     # patch from https://raw.githubusercontent.com/NixOS/nixpkgs/master/pkgs/development/python-modules/pygobject/pygobject-2.28.6-gio-types-2.32.patch
     # for https://bugzilla.gnome.org/show_bug.cgi?id=668522
     patch('pygobject-2.28.6-gio-types-2.32.patch', when='@2.28.6')
 
+    # pygobject version 2 requires an autotools build
```

Comment:
If we're dealing with multiple build systems it might be better to subclass from `Package`.

Reply:
@adamjstewart the older version (2.28.*) are autotools packages and my guess is that moving forward this will be a python package. My thought was that it would be better to make the spack package a pythonpackage for when/if we deprecate the older versions. This would save adding all of the python build logic that is built in to the pythonpackage. If you feel strongly about making it subclass Package, then I can do that, though.